app/services/importers/ok_sos/base_importer.rb

```diff
@@ -84,24 +90,24 @@ def parse_date(date)
         end
       end
 
-      def model_cache(klass, key)
-        cache_key = klass.to_s
-        return @model_cache[cache_key] if @model_cache[cache_key]
+      def id_cache(klass, key)
+        klass_key = klass.to_s
+        return @@id_cache[klass_key] if @@id_cache[klass_key]
 
-        @model_cache[cache_key] = klass.all.to_h { |x| [x[key].to_s, x] }
-        @model_cache[cache_key]
+        @@id_cache[klass_key] = klass.pluck(key, :id).to_h { |x| [x[0].to_s, x[1]] }
+        @@id_cache[klass_key]
       end
 
-      def get_cached(klass, key, value, create: false)
+      def lookup_cached_id(klass, key, value, create: false)
         return nil unless value.present? && value != '0'
 
-        return model_cache(klass, key)[value.to_s] if model_cache(klass, key)[value.to_s]
+        return id_cache(klass, key)[value.to_s] if id_cache(klass, key)[value.to_s]
 
-        raise ActiveRecord::RecordNotFound unless create
+        return nil unless create
```

Comment:
Return nil instead of raising an error if the record isn't found. Since we're storing external ids to the table it's a bit better here not to raise an error so we can still create records and then see what ids are missing if that makes sense (this will only work if the association is nullable or else will fail anyhow)